File: Algorithms/Graphs/Python/NaiveStringMatch.py
def naive_string_matching(text, pattern):
    n = len(text)
    m = len(pattern)

    occurrences = []

    for i in range(n - m + 1):
        j = 0
        while j < m and text[i+j] == pattern[j]:
            j += 1

        if j == m:
            occurrences.append(i)

    return occurrences


text = "ABABDABACDABABCABAB"
pattern = "ABABC"

occurrences = naive_string_matching(text, pattern)
print("Occurrences found at indices:", occurrences)

File: Algorithms/Graphs/C/NaiveStringMatch.c
#include <stdio.h>
#include <string.h>

void naiveStringMatching(char *text, char *pattern)
{
    int n = strlen(text);
    int m = strlen(pattern);

    for (int i = 0; i <= n - m; i++)
    {
        int j;
        for (j = 0; j < m; j++)
        {
            if (text[i + j] != pattern[j])
                break;
        }

        if (j == m)
            printf("Pattern found at index %d\n", i);
    }
}

int main()
{
    char text[] = "ABABDABACDABABCABAB";
    char pattern[] = "ABABC";

    naiveStringMatching(text, pattern);

    return 0;
}
